dashboard/app/models/pd/application/teacher_application.rb

```diff
@@ -102,10 +102,12 @@ class TeacherApplication < ApplicationBase
     # based on these rules in order:
     # 1. Application has a specific school? always overwrite the user's school info
     # 2. User doesn't have a specific school? overwrite with the custom school info.
+    # Will only update the school info if we have enough information for it
+    # An incomplete application may not have all the information we need to update
     def update_user_school_info!
       if school_id || user.school_info.try(&:school).nil?
-        school_info = get_duplicate_school_info(school_info_attr) || SchoolInfo.create!(school_info_attr)
-        user.update_school_info(school_info)
+        school_info = school_info_attr
+        user.update_school_info(get_duplicate_school_info(school_info) || SchoolInfo.create!(school_info)) if school_info
```

Comment:
sorry to go into the weeds on style here, but I think this logic is getting a bit complicated for a trailing if. a pattern that's worked great for us in ruby for small functions like this one is to prefer early returns over conditionals and end up with something like this:
```suggestion
      return unless school_id || user.school_info.try(&:school).nil?
      school_info = school_info_attr
      return unless school_info
      user.update_school_info(get_duplicate_school_info(school_info) || SchoolInfo.create!(school_info))
```
when there is only one layer of `if` logic, early returns are about the same as conditionals. but the nice part is that this approach completely keeps you out of nested conditionals. it doesn't work great for large methods, but that can be a benefit too, because it forces you to to break up large methods into smaller ones.

of course there's no code.org style rule against conditionals or nested conditionals... and you could also just change the trailing if to
```
      if school_info
        user.update_school_info(get_duplicate_school_info(school_info) || SchoolInfo.create!(school_info))
      end
```
my reason for pushing on getting rid of the trailing `if` in this situation is that once you have a compound statement (due to the `||`) it gets harder to reason about what the trailing `if` is doing.

Reply:
Ah, this is very helpful––I'm not familiar with best style in Ruby, so this is appreciated. Will refactor!